src/main/java/com/atlassian/jira/cloud/jenkins/listeners/AutoDeploymentsListener.java

```diff
@@ -0,0 +1,72 @@
+package com.atlassian.jira.cloud.jenkins.listeners;
+
+import com.atlassian.jira.cloud.jenkins.common.service.IssueKeyExtractor;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepStartNode;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.PrintStream;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AutoDeploymentsListener implements SinglePipelineListener {
```

Comment:
This is the entry point to the new "send deployments automatically" feature. The listener is notified on each node in a pipeline and decides when to send a deployment event to Jira.

Actually, it delegates that task to `SinglePipelineSingleDeploymentListener`, which listens to specific deployment nodes that matched against the deployments regular expression.